utilities/transactions/pessimistic_transaction_db.cc

```diff
@@ -589,6 +589,11 @@ bool WritePreparedTxnDB::IsInSnapshot(uint64_t prep_seq,
   // TODO(myabandeh): read your own writes
   // TODO(myabandeh): optimize this. This sequence of checks must be correct but
   // not necessary efficient
+  if (prep_seq == 0) {
```

Comment:
This check is actually redundant, because it will be capture by the `max_evicted_seq < snapshot_seq` check as max_evicted_seq is always no less than 0. Adding the check just to make it more clear.